test/integration-test/src/utils.rs

```diff
@@ -0,0 +1,61 @@
+//! Utilities to run tests
+
+use std::{
+    ffi::CString,
+    os::fd::RawFd,
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use aya::netlink_set_link_up;
+use libc::if_nametoindex;
+use netns_rs::{get_from_current_thread, NetNs};
+
+pub struct Netns {
+    pub name: String,
+    pub old_ns: NetNs,
+    pub ns: Option<NetNs>,
+}
+
+impl Netns {
+    pub fn new() -> Self {
+        static COUNTER: AtomicU64 = AtomicU64::new(0);
+        let pid = std::process::id();
+        let name = format!("aya-test-{pid}-{}", COUNTER.fetch_add(1, Ordering::Relaxed));
+
+        // Create and enter netns
+        let ns = NetNs::new(&name).unwrap_or_else(|e| panic!("Failed to create netns {name}: {e}"));
+        ns.enter().expect("Failed to enter network namespace");
```

Comment:
nit: include the name in all the messages?

Reply:
Yup, this one slipped through. Done in 8ec9f73